### PR TITLE
Add support for WITHOUT ROWID tables

### PR DESCRIPTION
**Pull request by John-He-928**
See #358 

Added `withoutRowid` parameter to `@Entity` annotation, which appends `WITHOUT ROWID` to the CREATE TABLE statement if set to true.

Simple test cases included.

### CODE REVIEW

File: floor_generator/lib/value_object/entity.dart [hunk 1]
```diff
@@ -100,7 +105,8 @@ class Entity extends Queryable {
       primaryKey.hashCode ^
       foreignKeys.hashCode ^
       indices.hashCode ^
-      constructor.hashCode;
+      constructor.hashCode ^
+      withoutRowid.hashCode;
 
   @override
   String toString() {
```
Comment (mqus) [inline floor_generator/lib/value_object/entity.dart:112]: Please also change the `toString` method.

**Comment by mqus** [comment]
So first of all, thank you for your contribution and sorry for not looking at it sooner. I have mainly two minor issues with your code: The first is the `toString` method (see the comment above) and the second is the following:
I'm wondering if it would make sense to add a check for points 1 and 3 in the [sqlite docs](https://sqlite.org/withoutrowid.html#differences_from_ordinary_rowid_tables) , meaning that `WITHOUT ROWID` tables should have a primary key that is not auto-incremented.

Apart from that, I'm pretty happy with your changes. But @vitusortner  should probably take a look.

**Comment by John-He-928** [comment]
@mqus Thank you for the review. I will update the code :)

**Comment by vitusortner** [comment]
Thanks for your contribution and thanks for your patience!

**Pull request merged**